packages/snaps-controllers/src/snaps/SnapController.ts

```diff
@@ -815,6 +815,13 @@ type SnapControllerArgs = {
    * MetaMetrics event tracking hook.
    */
   trackEvent: TrackEventHook;
+
+  /**
+   * A hook that returns a promise that resolves when the onboarding has completed.
+   *
+   * @returns A promise that resolves when onboarding is complete.
+   */
+  ensureOnboardingComplete: () => Promise<void>;
```

Comment:
Can we not get this from the onboarding controller state?

Reply:
Unfortunately onboarding controller only exists on extension. On mobile the onboarding state is stored in Redux only, so we'll have to come up with a way to monitor that.